solutions/usaco-1064.mdx

```diff
@@ -108,4 +108,105 @@ int main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class StuckInARut {
+	static int[] xCoordinates;
+	static int[] yCoordinates;
+
+	// sorts eastward cows by their y value
+	static class EastComparator implements Comparator<Integer> {
+		public int compare (Integer first, Integer second) {
+			return yCoordinates[first] - yCoordinates[second];
+		}
+	}
+
+	// sorts northward cows by their x value
+	static class NorthComparator implements Comparator<Integer> {
+		public int compare (Integer first, Integer second) {
+			return xCoordinates[first] - xCoordinates[second];
+		}
+	}
+
+	public static void main (String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		int n = Integer.parseInt(br.readLine());
+
+		List<Integer> eastCows = new ArrayList<>();
+		List<Integer> northCows = new ArrayList<>();
+		xCoordinates = new int[n];
+		yCoordinates = new int[n];
+
+		for (int i = 0; i < n; i++) {  // read input
+			StringTokenizer line = new StringTokenizer(br.readLine());
+			String direction = line.nextToken();
+			if (direction.equals("E")) {
+				eastCows.add(i);
+			}
+			else {
+				northCows.add(i);
+			}
+			xCoordinates[i] = Integer.parseInt(line.nextToken());
+			yCoordinates[i] = Integer.parseInt(line.nextToken());
+		}
+
+		// sort the east cows by y-coordinate
+		Collections.sort(eastCows, new EastComparator());
+		// sort the north cows by x-coordinate
+		Collections.sort(northCows, new NorthComparator());
+
+		boolean[] stopped = new boolean[n];
+		int[] amountStopped = new int[n];
+
+		for (int j : eastCows) {  // loop through all pairs of cows
+			for (int k : northCows) {
+				/* make sure that both the east and north cows 
+				 * are not stopped a collision is only possible 
+				 * if the x of the north cow is greater
+				 * than the x of east cow and the y of the east 
+				 * cow is greater than the y of the north cow
+				 */
+
+				if ((!stopped[j]) && (!stopped[k]) &&
+						(xCoordinates[k] > xCoordinates[j])
+						&& (yCoordinates[j] > yCoordinates[k])) {
+					if ((xCoordinates[k] - xCoordinates[j]) >
+							(yCoordinates[j] - yCoordinates[k])) {
+						// the north cow stops the east cow
+						// mark the east cow as stopped 
+						stopped[j] = true;
+
+						// add number of cows that the east cow stops + 1
+						amountStopped[k] += (1 + amountStopped[j]);
+					}
+					else if ((xCoordinates[k] - xCoordinates[j]) <
+							(yCoordinates[j] - yCoordinates[k])) {
+						// the east cow stops the north cow
+						// mark the north cow as stopped 
+						stopped[k] = true;
+
+						// add number of cows that the north cow stops + 1
+						amountStopped[j] += (1 + amountStopped[k]);
+					}
+					// if difference in x and difference in y is equal,
+					// let the cows continue as stated in the problem
```

Comment:
which...line of code is this supposed to correlate with?

Reply:
its like if (x > y), else if (x < y). if x is equal to y, then then the cows can continue moving because they are not blocked